executor/src/witgen/machines/mod.rs

```diff
@@ -42,14 +41,14 @@ pub trait Machine<'a, T: FieldElement>: Send + Sync {
     ) -> Option<EvalResult<'a, T>>;
 
     /// Returns the final values of the witness columns.
-    fn take_witness_col_values(&mut self, fixed_data: &'a FixedData<T>) -> HashMap<String, Vec<T>>;
+    fn take_witness_col_values(&mut self) -> HashMap<String, Vec<T>>;
 }
 
 /// All known implementations of [Machine].
 /// This allows us to treat machines uniformly without putting them into a `Box`,
 /// which requires that all lifetime parameters are 'static.
 pub enum KnownMachine<'a, T: FieldElement> {
-    SortedWitnesses(SortedWitnesses<T>),
+    SortedWitnesses(SortedWitnesses<'a, T>),
```

Comment:
why is this needed now?

Reply:
This machine was the only one that actually used fixed data but didn't already store a reference to it. Now, it stores a reference of type `&'a FixedData<'a, T>` (like other machines), so it needs a lifetime parameter itself.